user-facing-messages.md

```diff
@@ -0,0 +1,432 @@
+
+ERROR:    Application startup failed. Exiting.
+```
+
+### tests/fixtures/project_toml_salesforce_table_missing
+
+In the context of the production deploy self-check:
+
+```term
+Function failed validation: project.toml is missing the required '[com.salesforce]' table.
+```
+
+In the context of the CLI start command:
+
+```term
+Starting sf-functions-python v0.4.0 in single process mode.
+INFO:     Started server process [81740]
+INFO:     Waiting for application startup.
+ERROR:    RuntimeError: Unable to load function: project.toml is missing the required '[com.salesforce]' table.
+
+ERROR:    Application startup failed. Exiting.
+```
+
+### tests/fixtures/project_toml_salesforce_table_wrong_type
+
+In the context of the production deploy self-check:
+
+```term
+Function failed validation: project.toml is missing the required '[com.salesforce]' table.
+```
+
+In the context of the CLI start command:
+
+```term
+Starting sf-functions-python v0.4.0 in single process mode.
+INFO:     Started server process [81742]
+INFO:     Waiting for application startup.
+ERROR:    RuntimeError: Unable to load function: project.toml is missing the required '[com.salesforce]' table.
+
+ERROR:    Application startup failed. Exiting.
+```
+
+## Functions that fail at runtime
+
+These are cases we cannot catch using the self-check, as they only occur when the function is running.
+
+### tests/fixtures/raises_exception_at_runtime
+
+Server log:
+
+```term
+Starting sf-functions-python v0.4.0 in single process mode.
+INFO:     Started server process [81743]
+INFO:     Waiting for application startup.
+INFO:     Application startup complete.
+INFO:     Uvicorn running on http://localhost:8080 (Press CTRL+C to quit)
+Traceback (most recent call last):
+  File "/Users/emorley/src/sf-functions-python/salesforce_functions/_internal/app.py", line 79, in _handle_function_invocation
+    function_result = await function(event, context)
+                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+  File "/Users/emorley/src/sf-functions-python/tests/fixtures/raises_exception_at_runtime/main.py", line 5, in function
+    return 1 / 0
+           ~~^~~
+ZeroDivisionError: division by zero
+invocationId=00DJS0000000123ABC-fa35b79dc4047411826d3c52071066b9 level=error msg="Exception occurred whilst executing function: ZeroDivisionError: division by zero"
+```
+
+### tests/fixtures/return_value_not_serializable
+
+Server log:
+
+```term
+Starting sf-functions-python v0.4.0 in single process mode.
+INFO:     Started server process [81754]
+INFO:     Waiting for application startup.
+INFO:     Application startup complete.
+INFO:     Uvicorn running on http://localhost:8080 (Press CTRL+C to quit)
+invocationId=00DJS0000000123ABC-0baf07e3d0f724a9b944165a86e51f89 level=error msg="Function return value cannot be serialized: TypeError: Type is not JSON serializable: set"
```

Comment:
```suggestion
invocationId=00DJS0000000123ABC-0baf07e3d0f724a9b944165a86e51f89 level=error msg="Function return value can't be serialized: TypeError: Type is not JSON serializable: set."
```

Reply:
Same as https://github.com/heroku/sf-functions-python/pull/78/files#r1093776099 for the trailing full stop.